sybil/document.py

```diff
@@ -31,7 +31,7 @@ class Document:
     #: including not executing the example at all, modifying the :class:`~sybil.example.Example`
     #: or the :class:`~sybil.document.Document`, or calling the original evaluator on the example.
     #: This last case should always take the form of ``example.region.evaluator(example)``.
-    evaluator: Evaluator = None
+    evaluator: Optional[Evaluator] = None
```

Comment:
I'm torn - for me, the typing is correct before, I disagree with mypy's choice that "implicit optional is bad" - it's pretty explicit, to me, that the original `evaluator: Evaluator = None` is explicitly stating `evaluator` may be `None` or `Evaluator`.

The Python 3.11+ spelling of this makes mypy's position more obviously silly:

```python
evaluator: Evaluator | None = None
```

What would the impact be, if any, if we have mypy's config in this project turn "implicit optional" on?

Reply:
> The Python 3.11+ spelling of this makes mypy's position more obviously silly

It isn't "obviously silly" to me - it seems that this is a matter of personal preference. In particular, to me, not knowing the type of `Evaluator`, the original looks like `None` may be a valid `Evaluator`, when in fact it is not - this variable can be an `Evaluator` *or* `None`, and that's what the new type makes explicit.

> What would the impact be, if any, if we have mypy's config in this project turn "implicit optional" on?

Immediately fewer `mypy` errors, at the cost of the ambiguity mentioned above.

Please let me know how you would like me to proceed.

Reply:
> the original looks like `None` may be a valid `Evaluator`,  when in fact it is not
`None` is absolutely a valid value for the `evaluator` variable.

But whatever, I disagree with Python's typing choices here, where I feel `evaluator: Evaluator = None` is a perfectly explicit piece of syntax to cover this, however, it feels that's the way the language is going, so whatever, let's go it with it.